.github/actions/install-end2end-dependencies/action.yaml

```diff
@@ -8,9 +8,33 @@ runs:
     - name: Install dependencies
       shell: bash
       run: |-
+          function retry() {
+            local retries=$1
+            local delay=$2
+            shift 2
+
+            local count=1
+            while [ $count -le "$retries" ]; do
+              if ( "$@" ); then
+                return 0
+              fi
+
+              echo "Attempt $count/$retries failed. Retrying in $delay seconds..."
+              sleep "$delay"
+              count=$(( count + 1 ))
+            done
+
+            echo "All attempts failed. Exiting..."
+            exit 1
+          }
+
+          function retry_apt_get() {
+            retry 5 5 sudo apt-get "$@"
+          }
+
           set -exu;
-          sudo apt-get update
-          sudo apt-get install -y \
+          retry_apt_get update
+          retry_apt_get install -y \
```

Comment:
would  these dependencies not also encounter the 'azure.*' mirror issue as well?